service/mantle/fedex/FedexServices.xml

```diff
@@ -0,0 +1,240 @@
+                <set field="packageRouteSeg.trackingCode" from="responseMap.output.transactionShipments[0].masterTrackingNumber"/>
+                <set field="packageRouteSeg.labelDate" from="responseMap.output.transactionShipments[0].pieceResponses[0].deliveryDatestamp"/>
+                <set field="packageRouteSeg.labelUrl" from="responseMap.output.transactionShipments[0].pieceResponses[0].packageDocuments[0].url"/>
+                <entity-update value-field="packageRouteSeg"/>
+            </iterate>
+
+        </actions>
+    </service>
+    <service verb="make" noun="ShipmentInfo">
+       <in-parameters>
+           <parameter name="shipmentId" required="true"/>
+           <parameter name="shipmentRouteSegmentSeqId" required="true"/>
+           <parameter name="shippingGatewayConfigId" required="true"/>
+           <parameter name="shipment" type="EntityValue" required="true"/>
+           <parameter name="routeSegment" type="EntityValue" required="true"/>
+           <parameter name="shipmentMethodEnumId"/>
+       </in-parameters>
+       <out-parameters>
+           <parameter name="address_from" type="Map"/>
+           <parameter name="address_to" type="Map"/>
+           <parameter name="shipmentMethodEnumId"/>
+       </out-parameters>
+       <actions>
+
+           <!--Searching Origin and Dest PostalAddress-->
+           <entity-find-one entity-name="PostalAddress" value-field="originPostalAddress">
+               <field-map field-name="contactMechId" from="routeSegment.originPostalContactMechId"></field-map>
+           </entity-find-one>
+           <entity-find-one entity-name="PostalAddress" value-field="destPostalAddress">
+               <field-map field-name="contactMechId" from="routeSegment.destPostalContactMechId"></field-map>
+           </entity-find-one>
+
+           <!--Searching Origin and Dest  Organization -->
+           <entity-find-one entity-name="Organization" value-field="originOrganization">
+               <field-map field-name="partyId" from="shipment.fromPartyId"></field-map>
+           </entity-find-one>
+           <entity-find-one entity-name="Organization" value-field="destOrganization">
+               <field-map field-name="partyId" from="shipment.toPartyId"></field-map>
+           </entity-find-one>
+
+           <!--Searching Origin and Dest TelecomNumber -->
+           <entity-find-one entity-name="TelecomNumber" value-field="originNumber">
+               <field-map field-name="contactMechId" from="routeSegment.originTelecomContactMechId"></field-map>
+           </entity-find-one>
+           <entity-find-one entity-name="TelecomNumber" value-field="destNumber">
+               <field-map field-name="contactMechId" from="routeSegment.destTelecomContactMechId"></field-map>
+           </entity-find-one>
+
+           <!--Searching Origin and Dest of Organization -->
+           <entity-find-one entity-name="Person" value-field="originPerson">
```

Comment:
We can perform find on an OOTB view entity to find party details, instead of making 2 queries on Organization first and then Party, please explore, then we can discuss more on this.